pkg/data/domain/organization/common.go

```diff
@@ -0,0 +1,168 @@
+package organization
+
+import (
+	"context"
+	"sort"
+
+	securityv1alpha1 "github.com/giantswarm/apiextensions/v6/pkg/apis/security/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (s *Service) Get(ctx context.Context, options GetOptions) (Resource, error) {
+	var resource Resource
+	var err error
+
+	if len(options.Name) > 0 {
+		resource, err = s.getByName(ctx, options.Name)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	} else {
+		resource, err = s.getAll(ctx)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return resource, nil
+}
+
+func (s *Service) getByName(ctx context.Context, name string) (Resource, error) {
+	var err error
+
+	o := &Organization{}
+	{
+		org := &securityv1alpha1.Organization{}
+		err = s.client.CtrlClient().Get(ctx, client.ObjectKey{
+			Name:      name,
+			Namespace: metav1.NamespaceNone,
+		}, org)
+
+		if apierrors.IsNotFound(err) {
+			return nil, microerror.Mask(notFoundError)
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		o.Organization = omitManagedFields(org)
+		o.Organization.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   securityv1alpha1.SchemeGroupVersion.Group,
+			Version: securityv1alpha1.SchemeGroupVersion.Version,
+			Kind:    "Organization",
+		})
+	}
+
+	return o, nil
+}
+
+func (s *Service) getAll(ctx context.Context) (Resource, error) {
+	accessReview, err := s.getSelfSubjectAccessReview(ctx)
```

Comment:
First we check the SelfSubjectAccessReview to see if the user has permissions to list Organization CRs.

If the user does have sufficient permissions, then we load the list of Organization CRs

Otherwise we use SelfSubjectResourceRules to get a list of resources available to the user, we filter out names of resources of the "Organization" type, and we pull CR for each individual org name, which are visible to the user